test/e2e/data/atlascluster_serverless.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: atlas.mongodb.com/v1
+kind: AtlasCluster
+metadata:
+  name: atlascluster-serverless
+spec:
+  projectRef:
+    name: my-project
+  serverlessSpec:
+    name: serverless-instance
+    providerSettings:
+      providerName: SERVERLESS
```

Comment:
I think it is better to test with empty `backingProviderName` or test both variants 🤔

Reply:
While standing in the line realised this can be done at a later stage)